### PR TITLE
Updating to the k8s label convention

### LINKED ISSUE #4335: Update to Kubernetes recommended labels
A set of recommended labels around applications just landed in the Kubernetes docs. They can be seen at https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/. Helm also has a set of standard labels in the best practices section at https://docs.helm.sh/chart_best_practices/#standard-labels. Many of these are for the same thing but with a different name.

Here's a diff of the two:

| Helm | k8s |
| --- | --- |
| `heritage` | `app.kubernetes.io/managed-by` |
| `release` | `app.kubernetes.io/instance` |
| `chart` | - |
| `app` | - |
| `component` | `app.kubernetes.io/component` |
| - | `app.kubernetes.io/name` |
| - | `app.kubernetes.io/version` |
| - | `app.kubernetes.io/part-of` |

Note, the use of a prefix of `app.kubernetes.io` is because these are not private and we expect machines to do work with these. For those cases k8s uses the notion of namespacing them.

More details on the k8s ones are in the docs. With examples.

How should we handle this for Helm and Charts?

### PR DESCRIPTION
**Pull request by mattfarina**
Closes #4335

### CODE REVIEW

File: cmd/helm/testdata/testcharts/novals/templates/alpine-pod.yaml [hunk 1]
```diff
@@ -6,12 +6,12 @@ metadata:
     # The "heritage" label is used to track which tool deployed a given chart.
     # It is useful for admins who want to see what releases a particular tool
     # is responsible for.
-    heritage: {{.Release.Service | quote }}
+    app.kubernetes.io/managed-by: {{.Release.Service | quote }}
     # The "release" convention makes it easy to tie a release to all of the
     # Kubernetes resources that were created as part of that release.
-    release: {{.Release.Name | quote }}
+    app.kubernetes.io/instance: {{.Release.Name | quote }}
```
Comment (michelleN) [inline cmd/helm/testdata/testcharts/novals/templates/alpine-pod.yaml]: How come this one not `helm.sh/release`: {{.Release.Name | quote }}?

File: docs/chart_best_practices/labels.md [hunk 2]
```diff
@@ -25,8 +25,12 @@ are recommended, and _should_ be placed onto a chart for global consistency. Tho
...
-release | REC | This should be the `{{ .Release.Name }}`.
-chart | REC | This should be the chart name and version: `{{ .Chart.Name }}-{{ .Chart.Version \| replace "+" "_" }}`.
-app | REC | This should be the app name, reflecting the entire app. Usually `{{ template "name" . }}` is used for this. This is used by many Kubernetes manifests, and is not Helm-specific.
-component | OPT | This is a common label for marking the different roles that pieces may play in an application. For example, `component: frontend`.
+`app.kubernetes.io/name` | REC | This should be the app name, reflecting the entire app. Usually `{{ template "name" . }}` is used for this. This is used by many Kubernetes manifests, and is not Helm-specific.
+`helm.sh/chart` | REC | This should be the chart name and version: `{{ .Chart.Name }}-{{ .Chart.Version \| replace "+" "_" }}`.
+`app.kubernetes.io/managed-by` | REC | This should always be set to `{{ .Release.Service }}`. It is for finding all things managed by Tiller.
+`app.kubernetes.io/instance` | REC | This should be the `{{ .Release.Name }}`. It aid in differentiating between different instances of the same application.
```
Comment (michelleN) [inline docs/chart_best_practices/labels.md]: same question about it not being `helm.sh/release`

File: cmd/helm/testdata/testcharts/novals/templates/alpine-pod.yaml (hunk 1 above)
Comment (mattfarina) [reply cmd/helm/testdata/testcharts/novals/templates/alpine-pod.yaml]: A helm release provides a unique name for an instance of a chart. This allows me to do things like run multiple versions of the same chart where each has a unique name. That's the same purpose for the `app.kubernetes.io/instance` label. This is not a unique problem to helm.

This plays into the realm of compatibility with other tools, too.

Imagine dashboard wants to display all the different instances of applications no matter what deployed it. This is a common label to group and work on the things.

Does that help?

**Review by bacongobbler — APPROVED** [review]

File: cmd/helm/testdata/testcharts/novals/templates/alpine-pod.yaml (hunk 1 above)
Comment (michelleN) [reply cmd/helm/testdata/testcharts/novals/templates/alpine-pod.yaml]: Yea that makes sense!

**Review by michelleN — APPROVED** [review]

**Pull request merged**